rcongui/src/components/Blacklist/BlacklistRecords.js

```diff
@@ -15,10 +15,10 @@ import { BlacklistRecordCreateButton } from "./BlacklistRecordCreateDialog";
 import { Skeleton } from "@material-ui/lab";
 
 async function getBlacklistRecords(searchParams) {
-  let path = "get_blacklist_records?"
-  // remove all params = 0 or being falsy
-  let searchParamsString = Object.entries(searchParams).filter(param => param && param !== 0)
-  path += searchParamsString
+  let path = "get_blacklist_records?" + new URLSearchParams(
+    Object.entries(searchParams)
+      .filter(([_, v]) => v && v !== 0)
+  );
```

Comment:
Since the `blacklist_id` can be `0` and the user should be able to filter by the `default` blacklist the filter function needs to filter out all empty values only. However, it seems the server excludes expired blacklist records in either case when `exclude_expired = true OR exclude_expired = false` are provided. Therefore the filter needs to filter out false values as well.

```
  path += new URLSearchParams(
    Object.entries(searchParams).filter(
      ([_, value]) => value !== "" && value !== null && value !== undefined && value !== false
    )
  );
```

Reply:
![image](https://github.com/user-attachments/assets/7ce08489-d203-4f78-88ff-155d5de376c2)